jax/_src/random.py

```diff
@@ -2720,14 +2720,21 @@ def binomial(
   key, _ = _check_prng_key("binomial", key)
   check_arraylike("binomial", n, p)
   dtypes.check_user_dtype_supported(dtype)
-  if not dtypes.issubdtype(dtype, np.floating):
-    raise ValueError(
-        f"dtype argument to `binomial` must be a float dtype, got {dtype}"
-      )
+  if not (dtypes.issubdtype(dtype, np.floating) or dtypes.issubdtype(dtype, np.integer)):
+    raise ValueError(f"dtype argument to `binomial` must be a float or integer dtype, got {dtype}")
   dtype = dtypes.canonicalize_dtype(dtype)
+
   if shape is not None:
     shape = core.canonicalize_shape(shape)
-  return _binomial(key, n, p, shape, dtype)
+
+  float_type = dtypes.canonicalize_dtype(np.float32)
```

Comment:
Let's use `float_type = dtypes.to_inexact_dtype(dtype)` to respect the input dtype if it's floating point.